### PR TITLE
Improve and Document downstream usage of Selectable Colors - RSC-554

### PR DESCRIPTION
**Pull request by rpokorny**
https://issues.sonatype.org/browse/RSC-554

### CODE REVIEW

File: gallery/src/styles/SelectableColors/SelectableColorsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        include highlighted regions of a text document.
+      </NxP>
+      <NxP>
+        The colors that RSC provides for this purpose are known as the "selectable colors" and they are used within
+        the <NxCode>NxColorPicker</NxCode> and <NxCode>NxTag</NxCode> components as well as exposed publicly
+        for use in creating custom components.
+      </NxP>
+      <NxP>
+        For each named color within the selectable colors, two actual RGB color codes are provided, a lighter
+        variant and a darker one. This allows for two-tone components built with the selectable colors, as seen in
```
Comment (glenhunter) [inline gallery/src/styles/SelectableColors/SelectableColorsPage.tsx:38]:
```suggestion
        For each named color within the selectable colors, light and dark RGB color codes are provided. This allows for two-tone components built with the selectable colors, as seen in
```

File: gallery/src/styles/SelectableColors/SelectableColorsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    [selectableColorCustomExampleCode, { content: selectableColorCustomExampleStyles, language: 'scss' }];
+
+const SelectableColorsPage = () =>
+  <>
+    <GalleryDescriptionTile>
+      <NxP>
+        RSC provides a collection of color values which are intended to be used anywhere that the user is intended to
+        be able to select from a small set of colors and anywhere that a collection of arbitrarily but distinctly
+        colored elements is needed. Examples of the former include tags and labels. Examples of the latter
+        include highlighted regions of a text document.
```
Comment (glenhunter) [inline gallery/src/styles/SelectableColors/SelectableColorsPage.tsx:29]:
```suggestion
        RSC provides a palette of color values which are intended to be used for arbitrary but distinctly
        colored elements. Examples include components such as tags and labels, or user interface elements like 
        highlighted regions of a text document.
```

File: gallery/src/styles/SelectableColors/SelectableColorsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      </NxP>
+      <NxP>
+        The colors that RSC provides for this purpose are known as the "selectable colors" and they are used within
+        the <NxCode>NxColorPicker</NxCode> and <NxCode>NxTag</NxCode> components as well as exposed publicly
+        for use in creating custom components.
+      </NxP>
+      <NxP>
+        For each named color within the selectable colors, two actual RGB color codes are provided, a lighter
+        variant and a darker one. This allows for two-tone components built with the selectable colors, as seen in
+        unselected selectable <NxCode>NxTag</NxCode>s.
```
Comment (glenhunter) [inline gallery/src/styles/SelectableColors/SelectableColorsPage.tsx:39]:
```suggestion
        the selected and unselected states of selectable <NxCode>NxTag</NxCode>s.
```

File: gallery/src/styles/SelectableColors/SelectableColorsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        </NxTile.SubsectionHeader>
+        <NxP>
+          A number of CSS classes and custom CSS variables are provided to facilitate the use of the selectable
+          colors for styling custom components. Essentially, for each selectable color, a CSS class is provided
+          which defines two custom properties: <NxCode>--nx-selectable-color-light</NxCode>{' '}
+          and <NxCode>--nx-selectable-color-dark</NxCode>. Where a custom component wants to use a selectable
+          color, they would add that color's class to the element in question (or a parent) and then use the custom
+          properties to set styles as they see fit.  For instance, they might
+          use <NxCode>--nx-selectable-color-light</NxCode> to set the element's background while
+          using <NxCode>--nx-selectable-color-dark</NxCode> to set the text color.
```
Comment (glenhunter) [inline gallery/src/styles/SelectableColors/SelectableColorsPage.tsx:68]: I'm not sure that we should say this unless we're sure that these light/dark colour pairs also pass accessibility requirements for contrast (@dturner32)

File: gallery/src/styles/SelectableColors/SelectableColorsPage.tsx (hunk 3 above)
Comment (rpokorny) [reply gallery/src/styles/SelectableColors/SelectableColorsPage.tsx:37]: That suggestion is less accurate.  It is specifically the unselected state of selectable tags which demonstrates this.

File: gallery/src/styles/SelectableColors/SelectableColorsPage.tsx (hunk 4 above)
Comment (rpokorny) [reply gallery/src/styles/SelectableColors/SelectableColorsPage.tsx]: > We shouldn't say this unless we've confirmed that these light/dark colour pairs also pass accessibility requirements for contrast

Ah yeah I'm pretty certain they don't pass contrast requirements.  Changed to suggest background and border rather than background and text. 45dede15

File: gallery/src/styles/SelectableColors/SelectableColorsPage.tsx (hunk 3 above)
Comment (glenhunter) [reply gallery/src/styles/SelectableColors/SelectableColorsPage.tsx]: > That suggestion is less accurate. It is specifically the unselected state of selectable tags which demonstrates this.

To me it's the fact that there are two states built with the selectable colours which is the point, but regardless the bigger issue with the language (to me) is that there are three uses of "selectable" in that sentence which makes it hard to parse.

File: gallery/src/styles/SelectableColors/SelectableColorsPage.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        highlighted regions of a text document.
+      </NxP>
+      <NxP>
+        The colors that RSC provides for this purpose are known as the "selectable colors" and they are used within
+        the <NxCode>NxColorPicker</NxCode> and <NxCode>NxTag</NxCode> components as well as exposed publicly
+        for use in creating custom components.
+      </NxP>
+      <NxP>
+        For each named color within the selectable colors, light and dark RGB color codes are provided. This allows for
+        two-tone components built with the selectable colors, as seen in unselected selectable <NxCode>NxTag</NxCode>s.
```
Comment (glenhunter) [inline gallery/src/styles/SelectableColors/SelectableColorsPage.tsx:37]:
```suggestion
        For each named color within the selectable colors, light and dark RGB color codes are provided. This allows for 
        the building of two-tone components like the selectable <NxCode>NxTag</NxCode>s.
```

File: gallery/src/styles/SelectableColors/SelectableColorCustomExample.scss [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  // The values of these custom properties depend on which .nx-selectable-color--… class is applied
+  background-color: var(--nx-selectable-color-dark);
+  color: var(--nx-selectable-color-light);
+
+  border-radius: 12px / 8px;
+  display: inline-block;
+  height: 16px;
+  width: 24px;
+  line-height: 14px;
+  margin-right: 16px;
```
Comment (glenhunter) [inline gallery/src/styles/SelectableColors/SelectableColorCustomExample.scss:17]: Should probably use the variable here even if it is just an example

File: gallery/src/styles/SelectableColors/SelectableColorsPage.tsx (hunk 3 above)
Comment (rpokorny) [reply gallery/src/styles/SelectableColors/SelectableColorsPage.tsx]: > To me it's the fact that there are two states built with the selectable colours which is the point, but regardless the bigger issue with the language (to me) is that there are three uses of "selectable" in that sentence which makes it hard to parse.

How's this: b4836612

File: gallery/src/styles/SelectableColors/SelectableColorsPage.tsx (hunk 5 above)
Comment (rpokorny) [reply gallery/src/styles/SelectableColors/SelectableColorsPage.tsx]: Changed when addressing another comment. b483661

File: gallery/src/styles/SelectableColors/SelectableColorCustomExample.scss (hunk 6 above)
Comment (rpokorny) [reply gallery/src/styles/SelectableColors/SelectableColorCustomExample.scss]: > Should probably use the variable here even if it is just an example

f10f2910

**Review by glenhunter — APPROVED** [review]
